src/components/clusters/clusterPolling.ts

```diff
@@ -38,4 +42,6 @@ export const useClusterPolling = (clusterId: string) => {
       dispatch(cleanCluster());
     };
   }, [dispatch, fetchCluster]);
+
+  return [data, uiState, errorDetail];
```

Comment:
Nit: It may be slightly more convenient to return an object, wdyt?
```suggestion
  return {cluster: data, uiState, errorDetail};
```

Reply:
Array allows to name the variables in whatever way consumer likes without extra code

ie

```js
const [foo, bar, baz] = useClusterPolling(clusterId);
```

vs
```js
const {cluster: foo, uiState: bar, errorDetail: baz} = useClusterPolling(clusterId);
```

The benefit becomes obvious when you want to use `useClusterPolling` multiple times in one component so you need to name the constants differently.
We dont have such a case yet so I dont have a strong preference over object vs array. I'll go with whatever you prefer :)

Reply:
ok, yeah, object lets you pick items slightly easily but it is not needed in this case, so array it is

Reply:
I agree with @rawagner argument in this case it